osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -349,15 +350,23 @@ protected override void LoadComplete()
             });
         }
 
+        private static readonly LocalisableString input_search_placeholder = Resources.Localisation.Web.CommonStrings.InputSearch;
+        private static readonly LocalisableString tab_to_search_placeholder = ModSelectOverlayStrings.TabToSearch;
+
         protected override void Update()
         {
             base.Update();
 
-            SearchTextBox.PlaceholderText = SearchTextBox.HasFocus ? Resources.Localisation.Web.CommonStrings.InputSearch : ModSelectOverlayStrings.TabToSearch;
+            SearchTextBox.PlaceholderText = SearchTextBox.HasFocus ? input_search_placeholder : tab_to_search_placeholder;
 
             if (beatmapAttributesDisplay != null)
             {
-                float rightEdgeOfLastButton = footerButtonFlow.Last().ScreenSpaceDrawQuad.TopRight.X;
+                ShearedButton lastFooterButton = null!;
+
+                foreach (var b in footerButtonFlow)
```

Comment:
This is appalling legibility.

Surely we can at least use `footerButtonFlow[^1]`.